cli/generate.go

```diff
@@ -0,0 +1,279 @@
+	case "gnosis", "chiado":
+		vlStartGracePeriod = 2 * configs.EpochTimeGNO
+	default:
+		vlStartGracePeriod = 2 * configs.EpochTimeETH
+	}
+	// Generate docker-compose scripts
+	gd := generate.GenData{
+		ExecutionClient:         combinedClients.Execution,
+		ConsensusClient:         combinedClients.Consensus,
+		ValidatorClient:         combinedClients.Validator,
+		Network:                 network,
+		CheckpointSyncUrl:       flags.checkpointSyncUrl,
+		FeeRecipient:            flags.feeRecipient,
+		JWTSecretPath:           flags.jwtPath,
+		Graffiti:                flags.graffiti,
+		FallbackELUrls:          flags.fallbackEL,
+		ElExtraFlags:            flags.elExtraFlags,
+		ClExtraFlags:            flags.clExtraFlags,
+		VlExtraFlags:            flags.vlExtraFlags,
+		MapAllPorts:             flags.mapAllPorts,
+		Mev:                     !flags.noMev && utils.Contains(services, validator) && !flags.noValidator,
+		MevImage:                flags.mevImage,
+		LoggingDriver:           configs.GetLoggingDriver(logging),
+		RelayURL:                flags.relayURL,
+		MevBoostService:         utils.Contains(services, mevBoost),
+		MevBoostEndpoint:        flags.mevBoostUrl,
+		Services:                services,
+		VLStartGracePeriod:      uint(vlStartGracePeriod.Seconds()),
+		ExecutionApiUrl:         flags.executionApiUrl,
+		ExecutionAuthUrl:        flags.executionAuthUrl,
+		ConsensusApiUrl:         flags.consensusApiUrl,
+		ECBootnodes:             flags.customEnodes,
+		CCBootnodes:             flags.customEnrs,
+		CustomTTD:               flags.customTTD,
+		CustomChainSpecPath:     customNetworkConfigsData.ChainSpecPath,
+		CustomNetworkConfigPath: customNetworkConfigsData.NetworkConfigPath,
+		CustomGenesisPath:       customNetworkConfigsData.NetworkGenesisPath,
+		CustomDeployBlock:       flags.customDeployBlock,
+		CustomDeployBlockPath:   customNetworkConfigsData.NetworkDeployBlockPath,
+		MevBoostOnValidator:     flags.mevBoostOnVal,
+	}
+	err = sedgeAction.Generate(actions.GenerateOptions{
+		GenerationData: &gd,
+		GenerationPath: generationPath,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Print Env File
+	log.Infof(configs.CreatedFile, filepath.Join(generationPath, configs.DefaultEnvFileName))
+	if err = ui.PrintFileContent(out, filepath.Join(generationPath, configs.DefaultEnvFileName)); err != nil {
+		return err
+	}
+
+	// Print Docker Compose File
+	log.Infof(configs.CreatedFile, filepath.Join(generationPath, configs.DefaultDockerComposeScriptName))
+	if err = ui.PrintFileContent(out, filepath.Join(generationPath, configs.DefaultDockerComposeScriptName)); err != nil {
+		return err
+	}
+
+	// If teku is chosen, then prepare datadir with 777 permissions
+	if combinedClients.Consensus != nil && combinedClients.Consensus.Name == "teku" {
+		if err = preRunTeku(services, generationPath); err != nil {
+			return err
+		}
+	}
+	log.Info(configs.GenerationEnd)
+
+	return nil
+}
+
+func lessMevBoost(services []string) []string {
```

Comment:
Rename this function to `filterClients`, `onlyClients` or something more descriptive. You are getting only clients here

Reply:
Done at faacf787d646dade513226be2384704e0e6f9ad9